engine/class_modules/sc_death_knight.cpp

```diff
@@ -9844,11 +9896,28 @@ void death_knight_t::create_buffs()
         -> set_trigger_spell( conduits.eradicating_blow )
         -> set_cooldown( conduits.eradicating_blow -> internal_cooldown() );
 
+  buffs.meat_shield = make_buff( this, "meat_shield", find_spell( 338438 ) )
+        -> set_default_value( conduits.meat_shield.percent() )
+        -> set_trigger_spell( conduits.meat_shield )
+        -> set_stack_change_callback( [ this ]( buff_t*, int old_stacks, int new_stacks )
+          {
+            double ms_health = conduits.meat_shield.percent();
+            double health_change = ( 1.0 + new_stacks * ms_health ) / ( 1.0 + old_stacks * ms_health );
+
+            resources.initial_multiplier[ RESOURCE_HEALTH ] *= health_change;
+
+            recalculate_resource_max( RESOURCE_HEALTH );
+          } );
+
   buffs.unleashed_frenzy = make_buff( this, "unleashed_frenzy", conduits.unleashed_frenzy->effectN( 1 ).trigger() )
       -> add_invalidate( CACHE_STRENGTH )
       -> set_default_value( conduits.unleashed_frenzy.percent() );
 
   // Legendaries
+
+  buffs.crimson_rune_weapon = make_buff( this, "crimson_rune_weapon", find_spell( 334526 ) )
+      -> set_default_value( find_spell( 334526 ) -> effectN( 1 ).percent() );
+
```

Comment:
Just like Runic Corruption or the T21 set bonus that crimson rune weapon is ~~copied from~~ based on, you also have to update the rune regen coefficient whenever the buff's state changes and make sure the sim knows they affect regeneration rates with set_affects_regen( true ).
https://github.com/simulationcraft/simc/blob/legion-dev/engine/class_modules/sc_death_knight.cpp#L4449

```suggestion
      -> set_default_value( find_spell( 334526 ) -> effectN( 1 ).percent() )
      -> set_affects_regen( true )
      -> set_stack_change_callback( [ p ]( buff_t*, int, int )
           { p -> _runes.update_coefficient(); } );
```